ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+cluster_profile=/var/run/secrets/ci.openshift.io/cluster-profile
+cluster_name=${NAMESPACE}-${JOB_NAME_HASH}
+
+out=/tmp/secret/install-config.yaml
+mkdir "$(dirname "${out}")"
+
+cluster_variant=
+if [[ -e "${SHARED_DIR}/install-config-variant.txt" ]]; then
+    cluster_variant=$(<"${SHARED_DIR}/install-config-variant.txt")
+fi
+
+base_domain=
+if [[ -e "${SHARED_DIR}/install-config-base-domain.txt" ]]; then
+    base_domain=$(<"${SHARED_DIR}/install-config-base-domain.txt")
+else
+    case "${CLUSTER_TYPE}" in
+    aws) base_domain=origin-ci-int-aws.dev.rhcloud.com;;
+    azure) base_domain=ci.azure.devcluster.openshift.com;;
+    gcp) base_domain=origin-ci-int-gce.dev.openshift.com;;
+    esac
+fi
+
+echo "Installing from release ${RELEASE_IMAGE_LATEST}"
+
+expiration_date=$(date -d '4 hours' --iso=minutes --utc)
+ssh_pub_key=$(<"${cluster_profile}/ssh-publickey")
+pull_secret=$(<"${cluster_profile}/pull-secret")
+
+workers=3
+if [[ "${cluster_variant}" =~ "compact" ]]; then
+    workers=0
+fi
+
+case "${CLUSTER_TYPE}" in
```

Comment:
there's really not a lot of shared logic between the various platforms.  It may make more sense to have separate steps e.g. `steps-registry/conf/aws` that sets up a basic AWS `install-config.yaml`, etc.

Reply:
Also, then FIPS and other adjustments that are platform agnostic would come in as an additional config step after the platform-specific steps had laid in the groundwork.

Reply:
Good as follow-ups when this is in prod.

Reply:
We still need to think about how to support an IaaS-agnostic job. My initial idea was having a single set of steps that can support all cases based on `$CLUSTER_TYPE`, but now that I've implemented a test (and considering future expansions), this seems too restrictive.

We can split the script once we have a clearer resolution.